### ISSUE TITLE
Startup crash with Plone 4.3.1

### ISSUE THREAD

**Issue by glenfant** [collective/Products.PloneSoftwareCenter#8]
Hi,
I just add Products.PloneSoftwareCenter in a Plone 4.1.3 buildout, and starting up the instance show this:

```
2013-07-27 21:34:40 INFO ZServer HTTP server started at Sat Jul 27 21:34:40 2013
    Hostname: 0.0.0.0
    Port: 8080
2013-07-27 21:34:42 INFO DocFinderTab Applied patch version 1.0.5.
2013-07-27 21:34:43 WARNING PloneSoftwareCenter ExternalStorage N/A, falling back to AttributeStorage
2013-07-27 21:34:44 WARNING ZODB.blob (74953) Blob dir /Users/glenfant/zope/instances/plone43/var/blobstorage/ has insecure mode setting
Traceback (most recent call last):
  File "/Users/glenfant/zope/instances/plone43/eggs/Zope2-2.13.20-py2.7.egg/Zope2/Startup/run.py", line 76, in <module>
    run()
  File "/Users/glenfant/zope/instances/plone43/eggs/Zope2-2.13.20-py2.7.egg/Zope2/Startup/run.py", line 22, in run
    starter.prepare()
  File "/Users/glenfant/zope/instances/plone43/eggs/Zope2-2.13.20-py2.7.egg/Zope2/Startup/__init__.py", line 86, in prepare
    self.startZope()
  File "/Users/glenfant/zope/instances/plone43/eggs/Zope2-2.13.20-py2.7.egg/Zope2/Startup/__init__.py", line 262, in startZope
    Zope2.startup()
  File "/Users/glenfant/zope/instances/plone43/eggs/Zope2-2.13.20-py2.7.egg/Zope2/__init__.py", line 47, in startup
    _startup()
  File "/Users/glenfant/zope/instances/plone43/eggs/Zope2-2.13.20-py2.7.egg/Zope2/App/startup.py", line 118, in startup
    load_zcml()
  File "/Users/glenfant/zope/instances/plone43/eggs/Zope2-2.13.20-py2.7.egg/Zope2/App/startup.py", line 52, in load_zcml
    load_site()
  File "/Users/glenfant/zope/instances/plone43/eggs/Zope2-2.13.20-py2.7.egg/Zope2/App/zcml.py", line 46, in load_site
    _context = xmlconfig.file(site_zcml)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 649, in file
    include(context, name, package)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 352, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 706, in finish
    actions = self.handler(context, **args)
  File "/Users/glenfant/zope/instances/plone43/eggs/Zope2-2.13.20-py2.7.egg/OFS/metaconfigure.py", line 46, in loadProducts
    xmlconfig.include(_context, zcml, package=product)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 352, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 706, in finish
    actions = self.handler(context, **args)
  File "/Users/glenfant/zope/instances/plone43/eggs/z3c.autoinclude-0.3.4-py2.7.egg/z3c/autoinclude/zcml.py", line 104, in includePluginsDirective
    includeZCMLGroup(_context, info, filename)
  File "/Users/glenfant/zope/instances/plone43/eggs/z3c.autoinclude-0.3.4-py2.7.egg/z3c/autoinclude/zcml.py", line 30, in includeZCMLGroup
    include(_context, filename, includable_package)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 352, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 706, in finish
    actions = self.handler(context, **args)
  File "/Users/glenfant/zope/instances/plone43/eggs/z3c.autoinclude-0.3.4-py2.7.egg/z3c/autoinclude/zcml.py", line 53, in includeDependenciesDirective
    includeZCMLGroup(_context, info, 'meta.zcml')
  File "/Users/glenfant/zope/instances/plone43/eggs/z3c.autoinclude-0.3.4-py2.7.egg/z3c/autoinclude/zcml.py", line 30, in includeZCMLGroup
    include(_context, filename, includable_package)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 548, in include
    processxmlfile(f, context)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 380, in processxmlfile
    parser.parse(src)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 107, in parse
    xmlreader.IncrementalParser.parse(self, source)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
    self.feed(buffer)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xml/sax/expatreader.py", line 352, in end_element_ns
    self._cont_handler.endElementNS(pair, None)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/xmlconfig.py", line 359, in endElementNS
    self.context.end()
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 558, in end
    self.stack.pop().finish()
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 705, in finish
    args = toargs(context, *self.argdata)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 1397, in toargs
    args[str(name)] = field.fromUnicode(s)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/fields.py", line 137, in fromUnicode
    value = self.context.resolve(name)
  File "/Users/glenfant/zope/instances/plone43/eggs/zope.configuration-3.7.4-py2.7.egg/zope/configuration/config.py", line 179, in resolve
    mod = __import__(mname, *_import_chickens)
  File "/Users/glenfant/zope/instances/plone43/eggs/contentratings-1.0-py2.7.egg/contentratings/metaconfigure.py", line 4, in <module>
    from contentratings.category import RatingsCategoryFactory
  File "/Users/glenfant/zope/instances/plone43/eggs/contentratings-1.0-py2.7.egg/contentratings/category.py", line 10, in <module>
    from zope.app.container.contained import contained
zope.configuration.xmlconfig.ZopeXMLConfigurationError: File "/Users/glenfant/zope/instances/plone43/parts/instance/etc/site.zcml", line 16.2-16.23
    ZopeXMLConfigurationError: File "/Users/glenfant/zope/instances/plone43/eggs/Products.CMFPlone-4.3.1-py2.7.egg/Products/CMFPlone/configure.zcml", line 98.4-102.10
    ZopeXMLConfigurationError: File "/Users/glenfant/zope/instances/plone43/eggs/plone.contentratings-1.0.1_2-py2.7.egg/plone/contentratings/configure.zcml", line 9.2-9.37
    ZopeXMLConfigurationError: File "/Users/glenfant/zope/instances/plone43/eggs/contentratings-1.0-py2.7.egg/contentratings/meta.zcml", line 5.2-9.41
    ImportError: No module named container.contained
```

Using github's master and removing all other add-ons yields the same crash.
A mysterious bug because the "container.contained" don't appear in the contentratings/meta.zcml resoures ! (metaconfigure/metadirectives).

**Comment by davisagli** [comment]
It's not so mysterious once you remember ZCML tracebacks are a bit out of order: what's happening is contentratings' meta.zcml includes something from category.py, which imports from an old location. Probably needs to be updated to use zope.container instead of zope.app.container.

**Comment by glenfant** [comment]
Yes, I made some changes to make it work. Most of them are consequences of the changes in ZTK modules coming with Zope 2.13 (most zope.app.xxx features have merged into zope.xxx). And changes in the **contentratings** and **plone.app.contentratings** are required too.
Well I'm gonna checkout PSC, contentratings and plone.contentratings and checkin the changes in github. But IMHO, rating contents in PSC should be an optional feature, not a requirement.
OK, I'm gonna do the changes and try to support older Zope/Plone versions (at least from 4.0).

**Comment by glenfant** [comment]
OK, I made the required changes in the master branch of PSC but there is still work in contentratings + plone.contentratings.

**Comment by mauritsvanrees** [comment]
I have released contentratings 1.1 and plone.contentratings 1.1 today. They are compatible with Plone 4.3.
